File: java/src/patterns/creational/factory/abstractcreator/ParserCreator.java
package patterns.creational.factory.abstractcreator;

import patterns.creational.factory.productinterface.DocumentParser;

public abstract class ParserCreator {
    public abstract DocumentParser createParser();

    public void parse(String document) {
        DocumentParser parser = createParser();
        parser.parse(document);
    }
}

File: python/src/patterns/creational/factory/abstract_creator/parser_creator.py
from abc import ABC, abstractmethod
from src.patterns.creational.factory.product_interface.document_parser import DocumentParser

class ParserCreator(ABC):
    @abstractmethod
    def create_parser(self) -> DocumentParser:
        pass

    def parse(self, document: str) -> None:
        parser = self.create_parser()
        parser.parse(document)
